src/ZEO/cache.py

```diff
@@ -550,13 +553,22 @@ def load(self, oid):
     def loadBefore(self, oid, before_tid):
         noncurrent_for_oid = self.noncurrent.get(u64(oid))
         if noncurrent_for_oid is None:
-            self._trace(0x24, oid, "", before_tid)
-            return None
+            result = self.load(oid, before_tid)
+            if result:
+                return result[0], result[1], None
+            else:
+                self._trace(0x24, oid, "", before_tid)
```

Comment:
Probably, but that would be unrelated to this change.  The trace calls were just moved around a little.